packages/block-library/src/gallery/edit.js

```diff
@@ -227,7 +225,7 @@ class GalleryEdit extends Component {
 	componentDidMount() {
 		const { attributes, mediaUpload } = this.props;
 		const { images } = attributes;
-		if ( every( images, ( { url } ) => isBlobURL( url ) ) ) {
+		if ( Platform.OS === 'web' && every( images, ( { url } ) => isBlobURL( url ) ) ) {
```

Comment:
Since lodash `every` returns a "vacuous truth" for empty arrays, on mobile we encounter this when adding a new gallery block. The problem is that `mediaUpload` will be undefined, so we get a crash without that check.

Reply:
> Since lodash every returns a "vacuous truth" for empty arrays

In this case, could we add a check for empty arrays instead?

Reply:
Sure! Updated. :+1:

Reply:
But then, aren't we bringing the crash possibility back on mobile? What if the post was somehow saved with blob urls and opened in mobile?

Also, methods like "getBlobByURL, isBlobURL, revokeBlobURL, mediaUpload" doesn't make sense on mobile, so by adding the platform check we are making it obvious. Otherwise someone can change the condition in the future without knowing these shouldn't be called on mobile. wdyt?

Reply:
Good points @pinarol. I guess in the future we should extract this to a function and on the native version, we return false/do nothing to avoid these checks in the middle of the code. But for now, I'm ok with having the platform checks if this makes things more reliable.

Reply:
> in the future we should extract this to a function

That sounds like a good idea 👍 Let's iterate on it. For the current state it'll be safer to have the platform check.